code/__HELPERS/unsorted.dm

```diff
@@ -160,64 +160,59 @@ GLOBAL_REAL_VAR(list/stack_trace_storage)
 			return NORTHWEST
 
 
-
-/proc/LinkBlocked(turf/A, turf/B)
+/**
+ *	Returns true if the path from A to B is blocked. Checks both paths where the direction is diagonal
+ *	Variables:
+ *	bypass_window - whether it will go through transparent windows like lasers
+ *	projectile - whether throwpass will be checked to ignore dense objects like projectiles
+ *	bypass_xeno - whether to bypass dense xeno structures like flamers
+ */
+/proc/LinkBlocked(turf/A, turf/B, bypass_window = FALSE, projectile = FALSE, bypass_xeno = FALSE)
```

Comment:
Seems like you could just have a PASSX bitflag here instead of running a separate var for each thing to check.

Reply:
mesmoothbrain, what's that?

Reply:
Just a single var to use any combination of https://github.com/tgstation/TerraGov-Marine-Corps/blob/master/code/__DEFINES/equipment.dm#L1
You'd need to add a define for xeno walls, not 100% sure if any apply perfectly to what you want with bullets.